### ISSUE TITLE
Some occurrence fields cannot be downloaded

### ISSUE THREAD

**Issue by adam-collins** [AtlasOfLivingAustralia/downloads-plugin#83]

**Comment by adam-collins** [comment]
Custom occurrence download option does not download everything. e.g. outlierLayer

Current custom behaviour:
- `DARWIN CORE CLASS (FIELD GROUPS)` includes fields that have a matching DwC class (except for `environmental` and `contextual`). 
- `SPECIES TRAITS` includes fields `aust_conservation, state_conservation` and `species_group, species_subgroup`
- `SPATIAL INTERSECTION` includes `cl*` and `el*` layers.
- `MISC` `quality assertions` includes `assertions` separated into columns for a True/False value
- `MISC` `misc fields` previously included fields prefixed with `_` which is  all in `dynamicProperties` in biocache3, the catch all bucket for fields not indexed. 
- `MISC` `wild net` includes a specific species list `dr*` into the `extra` downloads parameter. This results in the additional fields in that species list being included in the download.

Changes required:
- Add `DARWIN CORE CLASS` toggles for `ResourceRelationship`, `MaterialSample`, `GeologicalContext`.
- Change behaviour of `misc` to include fields without a DwC class that are not included elsewhere. This excludes `stateConservation`, `countryConservation`, `speciesGroup`, `speciesSubgroup`, `el*`, `cl*`, `assertions`. `dynamicProperties` has a DwC class and should not be added when toggling `MISC` `misc`.

It is possible to assign a DwC class for non-DwC fields. This is done in this file `https://github.com/AtlasOfLivingAustralia/biocache-service/blob/develop/src/main/webapp/WEB-INF/messages.properties` in the existing override section:
```
# Dwc categories - this overrides DwcTerm.group
class.class=Taxon
class.id=Occurrence
class.identifierRole=Identification
class.organismID=Occurrence
class.scientific_name_addendum=Taxon
class.species=Taxon
class.type_status_qualifier=Identification
```

**Comment by timhicks-ala** [comment]
It's currently possible to see "Outlier for layer" as a column (`outlierLayer`) using the ALA Legacy Format download option, but there doesn't appear to be a way to include the `outlierLayerCount` field itself.

The issue of getting at outlierLayerCount has been raised in a helpdesk ticket: https://support.ehelp.edu.au/a/tickets/166137

**Comment by adam-collins** [comment]
Missing fields are addressed in 2 repositories:
https://github.com/AtlasOfLivingAustralia/downloads-plugin/pull/88
https://github.com/AtlasOfLivingAustralia/biocache-service/pull/807

`outlierLayerCount` will be available in the `custom` download when `include Misc` is selected.

**Comment by sughics** [comment]
Included in downloads-plugin release 4.0.0 and biocache-service release 3.2.0
